### PR TITLE
[Merged by Bors] - feat: strengthen `tendsto_tsum_of_dominated_convergence` a little

### PR DESCRIPTION
**Pull request by vbeffara**
Replace the uniform bound by an eventual uniform bound along the filter.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by loefflerd — COMMENTED** [review]
This looks good! I've left a couple of minor comments. I coded up Tannery's theorem because someone asked for it on zulip, and then I was a bit disappointed that nobody seemed to be using it, so I'm glad someone's found a use for it :smile:

File: Mathlib/Analysis/Normed/Group/Tannery.lean [hunk 1]
```diff
@@ -68,15 +69,16 @@ lemma tendsto_tsum_of_dominated_convergence {α β G : Type*} {𝓕 : Filter α}
     simpa only [sum_add_tsum_compl h_sum, eq_sub_iff_add_eq'] using hS.tsum_eq
   have h2 : Tendsto (∑ k in T, f · k) 𝓕 (𝓝 (T.sum g)) := tendsto_finset_sum _ (fun i _ ↦ hab i)
   rw [Metric.tendsto_nhds] at h2
-  refine (h2 (ε / 3) (by positivity)).mp (eventually_of_forall (fun n hn ↦ ?_))
-  rw [dist_eq_norm, ← tsum_sub (h_suma n).of_norm h_sumg.of_norm,
-    ← sum_add_tsum_compl (s := T) ((h_suma n).of_norm.sub h_sumg.of_norm),
+  filter_upwards [h2 (ε / 3) (by positivity), h_suma, h_bound] with n h2 h_suma h_bound
+  -- refine (h2 (ε / 3) (by positivity)).mp (eventually_of_forall (fun n hn ↦ ?_))
```
[the remark this answers is not in the document]
Comment (vbeffara) [reply Mathlib/Analysis/Normed/Group/Tannery.lean:73]: Ah yes it was the line from your code when I was adapting it. Removing.

File: Mathlib/Analysis/Normed/Group/Tannery.lean [hunk 2]
```diff
@@ -68,15 +69,16 @@ lemma tendsto_tsum_of_dominated_convergence {α β G : Type*} {𝓕 : Filter α}
     simpa only [sum_add_tsum_compl h_sum, eq_sub_iff_add_eq'] using hS.tsum_eq
   have h2 : Tendsto (∑ k in T, f · k) 𝓕 (𝓝 (T.sum g)) := tendsto_finset_sum _ (fun i _ ↦ hab i)
   rw [Metric.tendsto_nhds] at h2
-  refine (h2 (ε / 3) (by positivity)).mp (eventually_of_forall (fun n hn ↦ ?_))
-  rw [dist_eq_norm, ← tsum_sub (h_suma n).of_norm h_sumg.of_norm,
-    ← sum_add_tsum_compl (s := T) ((h_suma n).of_norm.sub h_sumg.of_norm),
+  filter_upwards [h2 (ε / 3) (by positivity), h_suma, h_bound] with n h2 h_suma h_bound
```
[the remark this answers is not in the document]
Comment (vbeffara) [reply Mathlib/Analysis/Normed/Group/Tannery.lean:72]: Sure, that's something I already did after submitting. Indeed, better to stay close to the previous version.

I do clobber `h_suma` and `h_bound` which might not be so clean? Kind of using `filter_upwards` like `specialize`.

**Comment by vbeffara** [comment]
There is a use for it in the PNT project :-) although, maybe it will make more sense in the end to use `continuousOn_tsum` ...

**Review by loefflerd — APPROVED** [review]
Looks good to me!

**Comment by eric-wieser** [comment]
bors merge

Thanks!

**Pull request closed without merging**